openmdao/utils/general_utils.py

```diff
@@ -977,21 +977,27 @@ def common_subpath(pathnames):
     return ''
 
 
-def _is_slice(indices):
+def _is_slicer_op(indices):
     """
-    Check if an array of indices contains a slice object.
+    Check if an array of indices contains a colon or ellipsis operator.
 
     Parameters
     ----------
     indices : ndarray
-        Dotted pathnames of systems.
+        Indices to check.
 
     Returns
     -------
     bool
-        Returns True if indices contains a slice.
+        Returns True if indices contains a colon or ellipsis operator.
     """
-    return any(isinstance(i, slice) for i in indices)
+    if isinstance(indices, Iterable):
+        if isinstance(indices, (tuple, list, range, str)):
+            return any(isinstance(i, slice) or i == ... for i in indices)
```

Comment:
Since ... (or `Ellipsis`) is a special constant (like `None`) would it be more appropriate to use `is` instead of `==` here?

Reply:
In Dymos I'm often passing ndarrays of type int as indices to things.  Something like
```
src_idxs = np.array([0, 2, 3], dtype=int)
connect('foo', 'bar', src_indices=om.slicer(src_idxs, ...))
```
Using a numpy array in this case breaks (confusion over ambiguity with any/all)
I recommend something like this to filter out ndarrays in the slicer
```
        if isinstance(indices, (tuple, list, range, str)):
            return any(isinstance(i, slice) or (not isinstance(i, Iterable) and i == ...) for i in indices)
        else:
            return any(isinstance(i, slice) or (not isinstance(i, Iterable) and i == ...) for i in indices.flatten())
```

and as a test

```
    def test_om_slice_with_indices_and_ellipsis_in_connect(self):
        class SlicerComp(om.ExplicitComponent):
            def setup(self):
                self.add_input('x', np.ones((3, 4)))
                self.add_output('y', 1.0)

            def compute(self, inputs, outputs):
                outputs['y'] = np.sum(inputs['x']) ** 2.0

        p = om.Problem()

        p.model.add_subsystem('indep', om.IndepVarComp('x', arr_large_4x4))
        p.model.add_subsystem('row134_comp', SlicerComp())

        idxs = np.array([0, 2, 3], dtype=int)

        p.model.connect('indep.x', 'row134_comp.x', src_indices=om.slicer[idxs, ...])

        p.setup()
        p.run_model()

        assert_near_equal(p['row134_comp.x'], arr_large_4x4[(0, 2, 3), ...])
        assert_near_equal(p['row134_comp.y'], np.sum(arr_large_4x4[(0, 2, 3), ...])**2)
```